### PR TITLE
Add UpdateTeam permission support

### PR DESCRIPTION
**Pull request by mithmatt**
### Describe your changes :

See #2950

This supports creation of a role and adding UpdateTeam permission to the role.
Any user that has this role can update team composition of users

#
### Type of change :

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

 Backend: @sureshms @harshach

### CODE REVIEW

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/util/JsonUtilsTest.java [hunk 1]
```diff
@@ -29,8 +29,8 @@
 import org.junit.jupiter.api.Test;
 import org.openmetadata.catalog.entity.teams.Team;
 
-@Slf4j
 /** This test provides examples of how to use applyPatch */
```
Comment (mithmatt) [inline catalog-rest-service/src/test/java/org/openmetadata/catalog/util/JsonUtilsTest.java:32]: This was being flagged as a dangling comment by the IDE

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/permissions/PermisssionsResourceTest.java [hunk 2]
```diff
@@ -92,6 +92,7 @@ void get_permissions(String username, Map<MetadataOperation, Boolean> expectedOp
                 put(MetadataOperation.UpdateOwner, Boolean.TRUE);
                 put(MetadataOperation.UpdateTags, Boolean.TRUE);
                 put(MetadataOperation.DecryptTokens, Boolean.TRUE);
+                put(MetadataOperation.UpdateTeam, Boolean.TRUE);
```
Comment (mithmatt) [inline catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/permissions/PermisssionsResourceTest.java:95]: Admin has access to UpdateTeam

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/Entity.java [hunk 3]
```diff
@@ -179,6 +179,11 @@ public static EntityReference withHref(UriInfo uriInfo, EntityReference ref) {
     return ref.withHref(href);
   }
 
+  public static boolean shouldHaveOwner(@NonNull String entityType) {
+    // Team does not have an owner. (yet?)
```
Comment (mithmatt) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/Entity.java:183]: We may need to introduce this someday...

Team owner => manager of the team

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/security/SecurityUtil.java [hunk 4]
```diff
@@ -150,14 +155,4 @@ public static String getPrincipalName(Map<String, String> authHeaders) {
...
     return target.request();
   }
-
-  /**
-   * Returns true if authentication is enabled.
-   *
-   * @param securityContext security context
-   * @return true if jwt filter based authentication is enabled, false otherwise
-   */
-  public static boolean isSecurityEnabled(SecurityContext securityContext) {
```
Comment (mithmatt) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/security/SecurityUtil.java:160]: @vivekratnavel  This function is no longer used - removing it in this PR

**Review by vivekratnavel — APPROVED** [review]
+1 LGTM

**Review by harshach — APPROVED** [review]

**Pull request merged**